res/controllers/common-controller-scripts.js

```diff
@@ -28,6 +28,54 @@ String.prototype.toInt = function() {
     return a;
 };
 
+/**
+ * Discriminates whether an object was created using the `{}` synthax.
+ *
+ * Returns true when was an object was created using the `{}` synthax.
+ * False if the object is an instance of a class like Date or Proxy or an Array.
+ *
+ * Object.isSimpleObject({}) // true
+ * Object.isSimpleObject(null) // false
+ * Object.isSimpleObject(undefined) // false
+ * Object.isSimpleObject(new Date) // false
+ * Object.isSimpleObject(new (class {})()) // false
+ * @param {any} obj Object to test
+ * @returns {boolean} true if obj was created using the `{}` or `new Object()` synthax, false otherwise
+ */
+Object.isSimpleObject = function(obj) {
+    return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+};
+
+/**
+ * Deeply merges 2 objects (Arrays and Objects only, not Map for instance).
+ * @param target {object | Array} Object to merge source into
+ * @param source {object | Array} Object to merge into source
+ */
+Object.deepMerge = function(target, source) {
+    if (target === source || target === undefined || target === null || source === undefined || source === null) {
+        return;
+    }
+
+    if (Array.isArray(target) && Array.isArray(source)) {
+        const objTarget = target.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
+        const objSource = source.reduce((acc, val, idx) => Object.assign(acc, {[idx]: val}), {});
+        Object.deepMerge(objTarget, objSource);
+        target.length = 0;
+        target.push(...Object.values(objTarget));
```

Comment:
this will probably not retain indicies when its been given an partially empty array (eg `let arr = new Array(2); arr[1] = 1`. But it think its very much an edge case we can ignore.

Reply:
I don't see why. `new Array(2)` should fill the array with `undefined's.

Reply:
Thank you. TIL.